src/scheduler.ts

```diff
@@ -130,6 +188,49 @@ export class Scheduler {
     return taskIdCodec.toString()
   }
 
+  /**
+   * getTimeAutomationFees
+   * @param action type
+   * @param executions
+   * @returns fee
+   */
+   async getTimeAutomationFees(
+    action: AutomationAction,
+    executions: number,
+  ) : Promise<number> {
+    const polkadotApi = await this.getAPIClient()
+    const resultCodec = await (polkadotApi.rpc as any).automationTime.getTimeAutomationFees(action, executions);
+    return resultCodec.toJSON() as unknown as number
+  }
+
+  /**
+   * calculateOptimalAutostaking
+   * @param principal
+   * @param collator
+   * @returns duration and apy result
+   */
+  async calculateOptimalAutostaking(
+    principal: number,
+    collator: string,
+  ) : Promise<AutostakingResult> {
+    const polkadotApi = await this.getAPIClient()
+    const resultCodec = await (polkadotApi.rpc as any).automationTime.calculateOptimalAutostaking(principal, collator)
+    return resultCodec.toPrimitive() as AutostakingResult
```

Comment:
Makes the `AutostakingResult.apy` human readable.  But...converts both fields to a string?  `toPrimitive()` returns the correct values:

```
  const polkadotApi = await scheduler.getAPIClient();
  const result = await polkadotApi.rpc.automationTime.calculateOptimalAutostaking(
    500000000000,
    '691Fmzb8rhYmBxLvaqYEUApK22s3o6eCzC4whDY7dZZ83YYQ' // collator on turing-staging
  )
  console.log(result.toPrimitive()) 
# { period: 16, apy: 7.367639521572666 }
```

but when it's casted to `AutostakingResult`, both fields become strings.

Reply:
whoa really? afaik, casting to a type _shouldn't_ do anything in typescript. that's just to satisfy the types in typescript and should be abstracted away when TS compiles down to JS